### PR TITLE
MIG-323 - Add logging improvements to migplan_controller

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
This adds logging improvements and kube events for status conditions to the Migplan_controller, follows the pattern applied in #698 for Migstorage_controller.

### CODE REVIEW

**Review by djwhatle — COMMENTED** [review]
@shubham-pampattiwar just so I'm clear, this PR just:

1. Adds the eventRecorder for status conditions
2. Adds end of reconcile logging for status conditions
3. Moves all const messages for migplan controller directly into where conditions are set

Anything else?

**Comment by shubham-pampattiwar** [comment]
> @shubham-pampattiwar just so I'm clear, this PR just:
> 
>     1. Adds the eventRecorder for status conditions
> 
>     2. Adds end of reconcile logging for status conditions
> 
>     3. Moves all const messages for migplan controller directly into where conditions are set
> 
> 
> Anything else?

@djwhatle In a nutshell yes. :)

File: pkg/controller/migplan/validation.go [hunk 1]
```diff
@@ -381,7 +345,8 @@ func (r ReconcileMigPlan) validateSourceCluster(plan *migapi.MigPlan) error {
 			Type:     SourceClusterNotReady,
 			Status:   True,
 			Category: Critical,
```
Comment (alaypatel07) [inline pkg/controller/migplan/validation.go:347]: Not related to this PR but `Reason` key is missing

File: pkg/controller/migplan/validation.go [hunk 2]
```diff
@@ -440,7 +406,8 @@ func (r ReconcileMigPlan) validateDestinationCluster(plan *migapi.MigPlan) error
 			Type:     DestinationClusterNotReady,
 			Status:   True,
 			Category: Critical,
```
Comment (alaypatel07) [inline pkg/controller/migplan/validation.go:408]: missing `Reason`

**Review by alaypatel07 — APPROVED** [review]
apart from minor nits, LGTM

File: pkg/controller/migplan/validation.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply pkg/controller/migplan/validation.go:352]: Added the `Reason` key.

**Review by djwhatle — APPROVED** [review]

File: pkg/controller/migplan/migplan_controller.go [hunk 3]
```diff
@@ -362,8 +369,10 @@ func (r *ReconcileMigPlan) planSuspended(plan *migapi.MigPlan) error {
 		plan.Status.SetCondition(migapi.Condition{
 			Type:     Suspended,
 			Status:   True,
+			Reason:   SuspendedState,
```
Comment (alaypatel07) [inline pkg/controller/migplan/migplan_controller.go:372]: hrm, I am not sure if `Type: Suspended` and `Reason: SuspededState` is helpful or redundant.

File: pkg/controller/migplan/gvk.go [hunk 4]
```diff
@@ -82,8 +82,9 @@ func reportGVK(plan *migapi.MigPlan, incompatibleMapping map[string][]schema.Gro
 		plan.Status.SetCondition(migapi.Condition{
 			Type:     GVKsIncompatible,
 			Status:   True,
+			Reason:   NotCompatible,
```
Comment (alaypatel07) [inline pkg/controller/migplan/gvk.go:85]: drop

File: pkg/controller/migplan/pvlist.go [hunk 5]
```diff
@@ -373,9 +372,11 @@ func (r *ReconcileMigPlan) getDestStorageClass(pv core.PersistentVolume,
 				plan.Status.SetCondition(migapi.Condition{
 					Type:     PvWarnNoCephAvailable,
 					Status:   True,
+					Reason:   NotAvailable,
```
Comment (alaypatel07) [inline pkg/controller/migplan/pvlist.go:375]: drop

File: pkg/controller/migplan/validation.go [hunk 6]
```diff
@@ -81,16 +82,21 @@ const (
...
-	NotFound      = "NotFound"
-	KeyNotFound   = "KeyNotFound"
-	NotDistinct   = "NotDistinct"
-	LimitExceeded = "LimitExceeded"
-	NotDone       = "NotDone"
-	Done          = "Done"
-	Conflict      = "Conflict"
-	NotHealthy    = "NotHealthy"
-	Incorrect     = "Incorrect"
+	NotSet         = "NotSet"
```
Comment (alaypatel07) [inline pkg/controller/migplan/validation.go:85]: restore const list back to what we had?

**Review by alaypatel07 — CHANGES_REQUESTED** [review]
PTAL at few `Reason`s to drop

File: pkg/controller/migplan/validation.go (hunk 6 above)
Comment (shubham-pampattiwar) [reply pkg/controller/migplan/validation.go:85]: Done

File: pkg/controller/migplan/validation.go [hunk 7]
```diff
@@ -81,16 +82,15 @@ const (
...
-	NotSet        = "NotSet"
-	NotFound      = "NotFound"
-	KeyNotFound   = "KeyNotFound"
-	NotDistinct   = "NotDistinct"
-	LimitExceeded = "LimitExceeded"
-	NotDone       = "NotDone"
-	Done          = "Done"
-	Conflict      = "Conflict"
-	NotHealthy    = "NotHealthy"
-	Incorrect     = "Incorrect"
```
Comment (alaypatel07) [inline pkg/controller/migplan/validation.go:93]: is this intentional?

Comment (shubham-pampattiwar) [reply pkg/controller/migplan/validation.go:93]: Yes the Incorrect const wasn't used before.

**Review by alaypatel07 — APPROVED** [review]
ACK

**Review by djwhatle — APPROVED** [review]
VISACK

**Pull request merged**